core/enumerator/product_spec.rb

```diff
@@ -14,9 +37,33 @@
       elems.should == [[1, "X"], [1, "Y"], [2, "X"], [2, "Y"]]
     end
 
+    it "accepts infinite enumerator and a finite enumerable" do
+      enum = Enumerator.product(1.., "A".."C")
+      enum.size.should == Float::INFINITY
+      enum.take(4).should == [[1, "A"], [1, "B"], [1, "C"], [2, "A"]]
+    end
+
```

Comment:
**minor:** Wondering what is a purpose of these specs?

```ruby
it "accepts infinite enumerator and a finite enumerable" do
it "accepts infinite enumerator and uknown enumerator" do
it "accepts finite enumerator and unknown anumerator" do
```
Do you want to check that combinations of different enumerators are handled well? Or is it enough to check that each of them is handled properly in separate test cases?

Reply:
Ah, TBH I am not sure they are useful but it's OK to have them. MRI tests is a white box testing, so they might test corner cases specific to their implementation...

> What do you mean by to check that each of them is handled properly?
I meant to check that `Enumerator.product` handles these kinds of enumerators well.